classes/mock/generator.php

```diff
@@ -721,6 +721,7 @@ protected function getParametersSignature(\reflectionMethod $method, $forceMockC
                     $parameterCode .= ' = ' . var_export($parameter->getDefaultValue(), true);
                     break;
 
+                case $parameter->allowsNull():
```

Comment:
I guess this is not totally correct. For instance, given the function `f` signature:

```php
function f(?int $x, int $y);
```

The generated mock will be:

```php
function f(?int $x = null, int $y);
```

which is incorrect. We cannot have a required parameter *after* an optional parameter.